From af52be88c689521364e06924e26c8989a075e0e3 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Fri, 28 Sep 2018 15:33:34 +0800
Subject: [PATCH] MLK-15975-3: ASoC: fsl_sai: The offset of fifo_off is changed

Commit 786c8bd56324 ("MLK-19734-3: dmaengine: imx-sdma: change
fifo offset of fifo_num") change the offset of fifo_off, so
the sai driver need to be updated.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
(cherry picked from commit c94ce8776e01f1f40a866d4da89603ab042dde0f)
---
 sound/soc/fsl/fsl_sai.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -726,19 +726,19 @@ static int fsl_sai_hw_params(struct snd_
 						FSL_SAI_MAXBURST_TX * pins;
 				if (sai->is_dsd)
 					sai->dma_params_tx.fifo_num = pins +
-					   (sai->dataline_off_dsd[tx] << 8);
+					   (sai->dataline_off_dsd[tx] << 4);
 				else
 					sai->dma_params_tx.fifo_num = pins +
-						(sai->dataline_off[tx] << 8);
+						(sai->dataline_off[tx] << 4);
 			} else {
 				sai->dma_params_rx.maxburst =
 					FSL_SAI_MAXBURST_RX * pins;
 				if (sai->is_dsd)
 					sai->dma_params_rx.fifo_num = pins +
-					  (sai->dataline_off_dsd[tx] << 8);
+					  (sai->dataline_off_dsd[tx] << 4);
 				else
 					sai->dma_params_rx.fifo_num = pins +
-						(sai->dataline_off[tx] << 8);
+						(sai->dataline_off[tx] << 4);
 			}
 		}
 
